motionplan/planManager.go

```diff
@@ -612,6 +634,60 @@ func (pm *planManager) goodPlan(pr *rrtPlanReturn, opt *plannerOptions) (bool, f
 	return false, solutionCost
 }
 
+func (pm *planManager) planToRRTGoalMap(plan Plan, goal spatialmath.Pose) (*rrtMaps, error) {
+	planNodes := make([]node, 0, len(plan))
+	// Build a list of nodes from the plan
+	for _, planStep := range plan {
+		conf, err := pm.frame.mapToSlice(planStep)
+		if err != nil {
+			return nil, err
+		}
+		planNodes = append(planNodes, newConfigurationNode(conf))
+	}
+
+	if pm.useTPspace {
+		// Fill in positions from the old origin to where the goal was during the last run
+		planNodesOld, err := rectifyTPspacePath(planNodes, pm.frame, spatialmath.NewZeroPose())
+		if err != nil {
+			return nil, err
+		}
+
+		// Figure out where our new starting point is relative to our last one, and re-rectify using the new adjusted location
+		oldGoal := planNodesOld[len(planNodesOld)-1].Pose()
+		pathDiff := spatialmath.PoseBetween(oldGoal, goal)
+		planNodes, err = rectifyTPspacePath(planNodes, pm.frame, pathDiff)
+		if err != nil {
+			return nil, err
+		}
+	}
+	// Fill in costs
+	for i := 1; i < len(planNodes); i++ {
+		cost := pm.opt().DistanceFunc(&ik.Segment{
+			StartConfiguration: planNodes[i-1].Q(),
+			StartPosition:      planNodes[i-1].Pose(),
+			EndConfiguration:   planNodes[i].Q(),
+			EndPosition:        planNodes[i].Pose(),
+			Frame:              pm.frame,
+		})
+		planNodes[i].SetCost(cost)
+	}
+
+	var lastNode node
+	goalMap := map[node]node{}
+	for i := len(planNodes) - 1; i >= 0; i-- {
```

Comment:
Why this loop and not just doing this in the above loop?

Reply:
Good point, I'll combine.